openstack/dns/v2/zones/requests.go

```diff
@@ -55,3 +55,108 @@ func Get(client *gophercloud.ServiceClient, zoneID string) (r GetResult) {
 	_, r.Err = client.Get(zoneURL(client, zoneID), &r.Body, nil)
 	return
 }
+
+// CreateOptsBuilder allows extensions to add additional attributes to the Update request.
+type CreateOptsBuilder interface {
+	ToZoneCreateMap() (map[string]interface{}, error)
+}
+
+// CreateOpts specifies the base attributes used to create a zone.
+type CreateOpts struct {
+	// Attributes are settings that supply hints and filters for the zone.
+	Attributes map[string]string `json:"attributes,omitempty"`
+
+	// Email contact of the zone.
+	Email string `json:"email,omitempty"`
+
+	// Description of the zone.
+	Description string `json:"description,omitempty"`
+
+	// Name of the zone.
+	Name string `json:"name,required"`
+
+	// Masters specifies zone masters if this is a secondary zone.
+	Masters []string `json:"masters,omitempty"`
+
+	// TTL is the time to live of the zone.
+	TTL int `json:"-"`
```

Comment:
No change necessary here, just a comment: `json:"ttl,omitempty"` should be fine here since 0 isn't allowed. That would clean up `ToZoneCreate/UpdateMap` a bit.